src/api-service/__app__/pool/__init__.py

```diff
@@ -78,6 +78,15 @@ def post(req: func.HttpRequest) -> func.HttpResponse:
     if isinstance(request, Error):
         return not_ok(request, context="PoolCreate")
 
+    if not can_modify_pools(req):
```

Comment:
Can we pull this check out into `main()`? This and `call_if_user` chould be part of a common `call_if_authorized` function.

Reply:
While I think that level of reorg should happen, IMO, that should be done as part of the larger RBAC feature we've discussed.

Reply:
That seems sensible.